tributors/utils/prompt.py

```diff
@@ -27,3 +27,32 @@ def request_input():
             lines.append(line)
 
     return "\n".join(lines)
+
+
+def choice_prompt(prompt, choices, choice_prefix=None, multiple=False):
+    """Ask the user for a prompt, and only return when one of the requested
+       options is provided.
+       Parameters
+       ==========
+       prompt: the prompt to ask the user
+       choices: a list of choices that are valid.   
+       multiple: allow multiple responses (separated by spaces)
+    """
+    choice = None
+    print(prompt)
+    get_input = getattr(__builtins__, "raw_input", input)
+
+    if not choice_prefix:
+        choice_prefix = "/".join(choices)
+    message = "[%s] : " % (choice_prefix)
+
+    while choice not in choices:
+        choice = get_input(message).strip()
+
+        # If multiple allowed, add selection to choices if includes all vaid
+        if multiple is True:
+            contenders = choice.strip().split(" ")
+            if all([x in choices for x in contenders]):
```

Comment:
nit picking: no need for `[]`, ie.
```suggestion
            if all(x in choices for x in contenders):
```
irrelevant performance wise here but in general - just using generators is better than creating a list first